[Kurtyjlee] ip

Kurtyjlee opened the pull request:
# CringeBot

CringeBot feels your life with cringe and frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,
download it from [here](https://github.com/nus-cs2103-AY2324S1/ip/pull/206).
1. double-click it.
2. add your tasks.
3. let it manage your tasks for you 😉
4. And it is FREE!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Kurtyjlee committed:
level-0

Kurtyjlee committed:
level-1, echo

Kurtyjlee committed:
level-2, Add, List

Kurtyjlee committed:
level-3, mark as done

Kurtyjlee committed:
level-3, added text for tasks in list

Kurtyjlee committed:
level-4, todos, events and deadlines

Kurtyjlee committed:
A-TextUiTesting

Kurtyjlee committed:
level-5, handle errors for invalid inputs:

Kurtyjlee committed:
level-6 delete

Kurtyjlee committed:
A-Enums, edited code and added enums

Kurtyjlee committed:
finalised changes ip week 2

Kurtyjlee committed:
Implement level-7, storing data into a txt file

Kurtyjlee committed:
store objects instead of string into .ser

Kurtyjlee committed:
Implement level-8, processing date values for deadline

Kurtyjlee committed:
Add TaskList, Storage and Ui for more oop

Kurtyjlee committed:
A-MoreOOP

Kurtyjlee committed:
A-Packages

Kurtyjlee committed:
edit packages

Kurtyjlee committed:
Merge remote-tracking branch 'remotes/upstream/add-gradle-support'

Kurtyjlee committed:
Merge remote-tracking branch 'remotes/upstream/add-gradle-support'

Kurtyjlee committed:
A-Gradle

Kurtyjlee committed:
A-JUnit, test for TaskList and Deadline

Kurtyjlee committed:
A-JavaDoc, create JavaDoc

Kurtyjlee committed:
A-CodingStandard

Kurtyjlee committed:
Level-9, find items

Kurtyjlee committed:
Implement slight changes

Kurtyjlee committed:
fix conflicts

Kurtyjlee committed:
fix conflicts between origin and branch-A-CodingStandard

Kurtyjlee committed:
Merge branch 'branch-Level-9'

Kurtyjlee committed:
Merge branch 'master'

Kurtyjlee committed:
fix level-9

Kurtyjlee committed:
Fix keep running bug

Singa-pirate commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,63 @@
+package duke;
+
+public class Parser {
+    enum modifyStatus {
```
I like the use of enums here which make the code very readable!

Singa-pirate commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,58 @@
+     *
+     * @return true if marked and false otherwise.
+     */
+    public boolean isMarked() {
```
perhaps you could change this method name to something more like a verb (e.g. checkIfMarked)

Singa-pirate commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,221 @@
+
+public class TaskList implements Serializable {
+
+    public ArrayList<Task> tasks;
```
I like how the variable name is in plural form for a list of tasks!

Singa-pirate commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,221 @@
+
+        this.tasks.add(newTask);
+
+        String sayWord = "Got it. I've added this task:\n"
```
perhaps this can be phrased to be more like a noun (e.g. response)

Singa-pirate commented on src/main/java/duke/Todo.java:
```diff
@@ -0,0 +1,26 @@
+package duke;
+
+class Todo extends Task{
+    private final char taskType = 'T';
```
I suggest this constant value can have a name in screaming snake case (e.g. TASKTYPE)

Singa-pirate reviewed (commented):
LGTM except for some nits for you to consider!

Kurtyjlee committed:
create barebones gui for CringeBot

Kurtyjlee committed:
refactor with packages

Kurtyjlee committed:
Add duke features to JavaFX gui

Kurtyjlee committed:
add JavaDocs

Kurtyjlee committed:
fix find items display bug

sp4ce-cowboy commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,63 @@
+     */
+    public static boolean parseCommands(String nextLine, TaskList tasks, Storage storage) {
+        String firstWord = nextLine.split(" ")[0];
+        try {
```
I like the use of switch-case statements instead of if-else statements here, makes the code look cleaner!

sp4ce-cowboy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,38 @@
+     * Constructor for Duke.
+     *
+     * @param filePath file path to the storage.
+     */
```
I like how clean your main class is

sp4ce-cowboy commented on src/main/java/duke/Main.java:
```diff
@@ -0,0 +1,40 @@
+            e.printStackTrace();
+        }
+    }
+}
```
LGTM

sp4ce-cowboy commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,74 @@
+/**
+ * Class for parsing commands made by the user.
+ */
+public class Parser {
```
I like your use of Enums here, makes the code more readable!

sp4ce-cowboy commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,74 @@
+package duke.parser;
```
I like how your parser class is organized into a different package.

sp4ce-cowboy commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,49 @@
+    @Override
+    public String toString() {
+        String checkBox = super.isMarked() ? "[X]" : "[ ]";
+        char taskType1 = 'D';
```
Would it be better if the variable name was `taskTypeDeadline` instead of `taskType1`, making it easier to differentiate from the String variable below of a similar variable name?
```suggestion
        char taskTypeDeadline = 'D';
```

sp4ce-cowboy commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,49 @@
+    public String toString() {
+        String checkBox = super.isMarked() ? "[X]" : "[ ]";
+        char taskType1 = 'D';
+        String taskType = String.format("[%c]", taskType1);
```
```suggestion
        String taskType = String.format("[%c]", taskTypeDeadline);
```

sp4ce-cowboy commented on src/main/java/duke/tasks/Event.java:
```diff
@@ -0,0 +1,34 @@
+    @Override
+    public String toString() {
+        String checkBox;
+        char taskType1 = 'E';
```
Similar to an above comment, would it better if the variable name was `taskTypeEvent` instead of `taskType1`?

```suggestion
        char taskTypeEvent = 'E';
```

sp4ce-cowboy commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,62 @@
+/**
+ * Super class for tasks.
+ */
+public class Task implements Serializable {
```
This class looks great!

sp4ce-cowboy commented on src/main/java/duke/tasks/TaskList.java:
```diff
@@ -0,0 +1,221 @@
+        Task newTask;
+
+        switch(task) {
+            case DEADLINE:
```
Would it be better if the case statements were indented at the same level as the switch statement, as per the course's coding standard?

sp4ce-cowboy commented on src/main/java/duke/tasks/Todo.java:
```diff
@@ -0,0 +1,34 @@
+    @Override
+    public String toString() {
+        String checkBox;
+        char taskType1 = 'T';
```
Similar to an above comment, would it be better to rename the variable to `taskTypeTodo` instead? 
```suggestion
        char taskTypeTodo = 'T';
```

sp4ce-cowboy commented on src/main/java/duke/tasks/Todo.java:
```diff
@@ -0,0 +1,34 @@
+    public String toString() {
+        String checkBox;
+        char taskType1 = 'T';
+        String taskType = String.format("[%c]", taskType1);
```
```suggestion
        String taskType = String.format("[%c]", taskTypeTodo);
```

sp4ce-cowboy commented on src/main/java/duke/ui/DialogBox.java:
```diff
@@ -0,0 +1,78 @@
+/**
+ * Custom DialogBox to store dialogs between users and CringeBot
+ */
+public class DialogBox extends HBox {
```
LGTM

sp4ce-cowboy commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,59 @@
+ * Class that handles the statements said by CringeBot
+ */
+public class Ui {
+    private static final String BAR = "____________________________________________________________";
```
I like how you have followed the convention for naming class constants!

sp4ce-cowboy commented on src/test/java/duke/DeadlineTest.java:
```diff
@@ -0,0 +1,36 @@
+
+import duke.tasks.Deadline;
+
+public class DeadlineTest {
```
Perhaps you could consider naming your Unit tests according to the required convention of `TestDescription_TestInput_ExpectedOutput"?

sp4ce-cowboy commented on src/test/java/duke/TaskListTest.java:
```diff
@@ -0,0 +1,55 @@
+
+import java.util.ArrayList;
+
+public class TaskListTest {
```
Other than the test naming convention I mentioned above, LGTM!

sp4ce-cowboy reviewed (commented):
Overall coding standard is well adhered to, apart from some minor nits (the Unit Test naming conventions). Otherwise, LGTM!

Kurtyjlee committed:
Change Duke class and duke package to CringeBot and cringebot

- Update the cringebot package and all its dependencies
- Update the CringeBot class and all its dependencies

Kurtyjlee committed:
Remove all mentions of Duke

- Remove all mentions of Duke and changed project to CringeBot

Kurtyjlee committed:
Refactor code and improve code quality
- Fix naming
- Fix bad indentation for switches
- Change overly long string operations
- Update JavaDocs comments

Kurtyjlee committed:
Implement checkstyle
- Change code to meet checkstyle requirements

Kurtyjlee committed:
Add assertions

- Add assertions to storage and inputs

Kurtyjlee committed:
Merge pull request #2

Improve code quality of CringeBot

Kurtyjlee committed:
merge master to branch-A-Assertions

Kurtyjlee committed:
Merge pull request #1

Add assertions to CringeBot

Kurtyjlee committed:
Add github actions CI

Kurtyjlee committed:
Merge pull request #3 from Kurtyjlee/branch-A-CI

Add gh actions CI

Kurtyjlee committed:
Fix github workflows dir

Kurtyjlee committed:
fix and refactor test files

Kurtyjlee committed:
Add recurring feature for deadline

Kurtyjlee committed:
Merge pull request #4

Add recurring feature for deadline

Kurtyjlee committed:
Insert test cases for tasks and parser

Kurtyjlee committed:
Insert test for exceptions and storage

Kurtyjlee committed:
Satisfy checkstyleMain and checkstyleTest

Kurtyjlee committed:
change storageTest

Kurtyjlee committed:
Update README.md

Kurtyjlee committed:
Merge pull request #8 from Kurtyjlee/A-MoreTesting

A-MoreTesting pr

Kurtyjlee committed:
fix issues

Kurtyjlee committed:
satisfy checkstyle

Kurtyjlee committed:
Merge pull request #10 from Kurtyjlee/5-sharing-ip-code-quality-feedback-for-kurtyjlee

5 sharing ip code quality feedback for kurtyjlee

Kurtyjlee committed:
add better gui

Kurtyjlee committed:
Merge pull request #11 from Kurtyjlee/branch-A-BetterGui

add better gui

Kurtyjlee committed:
Add user guide

Kurtyjlee committed:
Merge pull request #12 from Kurtyjlee/branch-A-UserGuide

Add user guide

Kurtyjlee committed:
Update readme.md

Kurtyjlee committed:
Shift Ui.png
